### PR TITLE
Remove DataParallel remnants from DDP doc

### PR DESCRIPTION
**Pull request by rohan-varma**
As @aazzolini pointed out, the docstring is incorrect and probably vestige from DP / single process multi device mode in DDP.

Fixes #ISSUE_NUMBER

### CODE REVIEW

**Review by aazzolini — APPROVED** [review]
Thanks!

File: torch/nn/parallel/distributed.py [hunk 1]
```diff
@@ -240,14 +240,14 @@ def post_hook(self, is_last_joiner: bool):
...
     ``torch.distributed`` package at the module level.
-
-    This container parallelizes the application of the given module by
-    splitting the input across the specified devices by chunking in the batch
-    dimension. The module is replicated on each machine and each device, and
-    each such replica handles a portion of the input. During the backwards
-    pass, gradients from each node are averaged.
-
-    The batch size should be larger than the number of GPUs used locally.
+
```
Comment (awgu) [inline torch/nn/parallel/distributed.py:243]: Lint error is real :/

**Comment by rohan-varma** [comment]
pytorchbot merge -f "only changing string"

**Pull request closed without merging**